File: ProyectoSpring/SpringBussinesOne/demo/demo/src/main/java/com/BussinesOne/demo/models/Dtos/Requests/FichajePostRequestDto.java
package com.BussinesOne.demo.models.Dtos.Requests;

import jakarta.validation.constraints.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Data
public class FichajePostRequestDto {
    @NotNull
    private Long empleadoId;

    @NotNull
    private LocalDate fecha;

    @NotNull
    private LocalTime horaEntrada;

    private LocalTime horaSalida;

    private Boolean tiempoReal = true;

    private LocalDateTime createdAt; // si se deja null, el Backend asigna ahora con @PrePersist

}

File: ProyectoAndroid/app/src/main/java/com/example/proyectobussinesone/ComponenteKotlinPrueba/models/FichajePostRequestDto.kt
package com.example.proyectobussinesone.ComponenteKotlinPrueba.models


import com.google.gson.annotations.SerializedName
import java.time.LocalDate
import java.time.LocalTime

data class FichajePostRequestDto(
    @SerializedName("empleadoId")
    val empleadoId: Long,

    // Debe coincidir con el formato ISO que acepta tu backend, p.ej. "2025-06-01"
    @SerializedName("fecha")
    val fecha: String,

    @SerializedName("horaEntrada")
    val horaEntrada: String,

    @SerializedName("horaSalida")
    val horaSalida: String?,

    @SerializedName("tiempoReal")
    val tiempoReal: Boolean = true
)